aws-kendra-index/src/main/java/software/amazon/kendra/index/Translator.java

```diff
@@ -162,7 +173,11 @@ static UpdateIndexRequest translateToUpdateRequest(final ResourceModel currModel
                             currModel.getDocumentMetadataConfigurations(),
                             prevDocumentMetadataConfiguration))
             .capacityUnits(translateToCapacityUnitsConfiguration(currModel.getCapacityUnits(), currModel.getEdition()))
-            .build();
+            .userTokenConfigurations(translateToSdkUserTokenConfigurations(currModel.getUserTokenConfigurations()));
+    if (currModel.getUserContextPolicy() != null) {
+      builder.userContextPolicy(currModel.getUserContextPolicy());
```

Comment:
np; do we need this if statement? Can we just have `.userContextPolicy(currModel.getUserContextPolicy())` above? If we can get rid of the check we wont have to worry about an extra branch to test (maybe we happen to cover both branches now but in the future that could change).

Reply:
Updated in :https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-kendra/pull/81

GitHubGitHub